bin/travis-encrypt-file

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+set -euo pipefail # STRICT MODE
+IFS=$'\n\t'       # http://redsymbol.net/articles/unofficial-bash-strict-mode/
+
+random-hex() {
+  head -c 10 /dev/urandom | xxd -p
+}
+
+NC='\033[0m'
+
+plain() {
+  echo -en $* 1>&2
+}
+
+blue() {
+  echo -en "\033[0;34m$*${NC}" 1>&2
+}
+
+yellow() {
+  echo -en "\033[0;33m$*${NC}" 1>&2
+}
+
+red() {
+  echo -en "\033[0;31m$*${NC}" 1>&2
+}
+
+gray() {
+  echo -en "\033[0;37m$*${NC}" 1>&2
+}
+
+nl() {
+  echo 1>&2
+}
+
+if ! git config travis.slug &>/dev/null; then
+  plain "Please define the "
+  yellow "travis.slug"
```

Comment:
Any reason not to have defined the colors as variables, and then just interpolated? Something like
```bash
cat <<-EOF
        Please define the ${yellow}travis.slug${nc} value using ${yellow}git config${nc}
        (or by running ${yellow}travis encrypt{$nc} on something). ...
EOF
```
Or, alternatively, keep the functions but write it like this?
```bash
cat <<-EOF
        Please define the $(yellow travis.slug) value using $(yellow git config)
        (or by running $(yellow travis encrypt) on something). ...
EOF
```
Speaking of scripts, I'm slowly going through scripts and ensuring they pass [shellcheck](https://github.com/koalaman/shellcheck), are formatted to [shfmt](https://github.com/mvdan/sh) tastes (with a few options), and I plan on adding unit testing with [osht](https://github.com/mvdan/sh) next.

Travis has shellcheck pre-installed, so it might be interesting to add it as a step.

Reply:
> Any reason not to have defined the colors as variables, and then just interpolated? 

I find this way a lot clearer. Or at least I did at first. It's gotten a bit bigger and the newlines make it hard to follow, so I probably might have interpolated had I realized it was going to get this big when I started.

> Travis has shellcheck pre-installed, so it might be interesting to add it as a step.

I'm a little leery of linting the formatting, since we don't do that with anything else right now. I'm fine with linting other things though.